zsl/application/service_application.py

```diff
@@ -1,53 +1,131 @@
 ------------------------------------------
 """
 from __future__ import unicode_literals
-from flask import Flask
+
 import os
+from typing import Callable
 
-from zsl.interface.importer import is_initialized
+from flask import Flask, Config
 from flask_injector import FlaskInjector
+from injector import Injector, Binder, singleton
+
+from zsl._state import set_current_app
+from zsl.utils.warnings import deprecated
+from zsl.application.initialization_context import InitializationContext
 
 
 class AtteqServiceFlask(Flask):
-    """
-    Atteq Service Flask application.
-    """
+    """Atteq Service Flask application."""
 
     VERSION = '1.1.4'
 
     def __init__(self, import_name, static_path=None, static_url_path=None,
                  static_folder='static', template_folder='templates',
-                 instance_path=None, instance_relative_config=False):
-        Flask.__init__(self, import_name, static_path, static_url_path,
-                       static_folder, template_folder, instance_path,
-                       instance_relative_config)
+                 instance_path=None, instance_relative_config=False,
+                 modules=None):
+
+        super(AtteqServiceFlask, self).__init__(import_name, static_path, static_url_path,
+                                                static_folder, template_folder, instance_path,
+                                                instance_relative_config)
+
         self._dependencies_initialized = False
+        self._is_initialized = False
+
+        self._injector = None
+        self._worker = None
+
+        self._configure()
+
+        if not modules:
+            from zsl.application.containers.core_container import CoreContainer
+            modules = CoreContainer.modules()
+
+        self._configure_injector(modules)
+
+        self._initialize()
+
+        self._dependencies_initialized = True
+
+    def _configure(self):
+        """Read the configuration from config files."""
         self.config.from_object('settings.default_settings')
+
         asl_settings = os.environ.get('ASL_SETTINGS')
         if asl_settings is not None:
             self.config.from_envvar('ASL_SETTINGS')
 
-    def initialize_dependencies(self, initialization_context):
-        from zsl.application.initializers import injection_views, injection_modules
-        self._initialization_context = initialization_context
-        self._flask_injector = FlaskInjector(self, injection_views + injection_modules)
-        self.set_injector(self._flask_injector.injector)
-        self.logger.debug("Injector configuration {0}, {1}.".format(injection_views, injection_modules))
+    def _initialize(self):
+        """Run the initializers."""
+        ctx = self.injector.get(InitializationContext)
+
+        ctx.initialize()
+
+    def _register(self):
+        """Register the current instance into application stack."""
+        set_current_app(self)
+
+    def _get_app_module(self):
```

Comment:
Yes the name is weird, it was the best I could think out, the app instance needs to inject its own instance and config, so I need an "injector module", but a callable works too. Do you have some better way?